File: sample1/src/main/java/com/dozeboy/sample1/MainActivity.java
package com.dozeboy.sample1;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.dozeboy.sample1.base.BaseSampleActivity;
import com.dozeboy.sample1.databinding.ActivityMainBinding;

/**
 * @author dozeboy
 * @date 2019/1/6
 */
public class MainActivity extends BaseSampleActivity<ActivityMainBinding> {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding.textHello.setOnClickListener(v -> {
            startActivity(new Intent(this, MainActivity.class));
        });
        ActivityMainBinding.inflate(getLayoutInflater());
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }
}

File: myappshell/src/main/java/com/dozeboy/android/template/MainActivity.kt
package com.dozeboy.android.template

import android.os.Bundle
import androidx.appcompat.app.ActionBarDrawerToggle
import com.dozeboy.android.template.base.BaseActivity

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupDrawer()
    }

    private fun setupDrawer() {
        val toggle = ActionBarDrawerToggle(this, drawer_layout, toolbar, R.string.drawer_open, R.string.drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
    }
}
